nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/avro/AvroReplacementDescriptor.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.runtime.avro;
+
+import java.util.Objects;
+
+import org.nuxeo.common.xmap.annotation.XNode;
+import org.nuxeo.common.xmap.annotation.XObject;
+
+/**
+ * The avro forbidden character replacement descriptor.<br>
+ * Avro allows alphanumeric characters and underscores in names.<br>
+ * Nuxeo Studio allows alphanumeric characters, underscores and dashes that have to be replaced by another symbol.<br>
+ * <br>
+ * The default contributions provide replacement for :<br>
+ * - "-" as "__dash__"<br>
+ * - ":" as "__colon__<br>
+ * - ";" as "__semicolon__"<br>
+ * - and with higher priority "__" as "____" to ensure no user string is wrongly replaced.<br>
```

Comment:
It seems quite overkill to make all this configurable through XML. I'd prefer a simple replacement method that hardcodes what we need.